aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255.move

```diff
@@ -0,0 +1,1271 @@
+
+module aptos_std::ristretto255 {
+    use std::option::Option;
+
+    #[test_only]
+    use std::option;
+
+    //
+    // Constants
+    //
+
+    /// The order of the Ristretto255 group and its scalar field, in little-endian.
+    const ORDER_ELL : vector<u8> = vector[
+        0xed, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58,
+        0xd6, 0x9c, 0xf7, 0xa2, 0xde, 0xf9, 0xde, 0x14,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10,
+    ];
+
+    /// The maximum size in bytes of a canonically-encoded Scalar is 32 bytes.
+    const MAX_SCALAR_NUM_BYTES : u64 = 32u64;
+
+    /// The maximum size in bits of a canonically-encoded Scalar is 256 bits.
+    const MAX_SCALAR_NUM_BITS : u64 = 256u64;
+
+    /// The maximum size in bytes of a canonically-encoded Ristretto255 point is 32 bytes.
+    const MAX_POINT_NUM_BYTES: u64 = 32u64;
+
+    /// The basepoint (generator) of the Ristretto255 group
+    const BASE_POINT: vector<u8> = vector[
+        0xe2, 0xf2, 0xae, 0x0a, 0x6a, 0xbc, 0x4e, 0x71,
+        0xa8, 0x84, 0xa9, 0x61, 0xc5, 0x00, 0x51, 0x5f,
+        0x58, 0xe3, 0x0b, 0x6a, 0xa5, 0x82, 0xdd, 0x8d,
+        0xb6, 0xa6, 0x59, 0x45, 0xe0, 0x8d, 0x2d, 0x76,
+    ];
+
+    //
+    // Reasons for error codes
+    //
+
+    /// The number of scalars does not match the number of points.
+    const E_DIFFERENT_NUM_POINTS_AND_SCALARS: u64 = 1;
+    /// Expected more than zero points as input.
+    const E_ZERO_POINTS: u64 = 2;
+    /// Expected more than zero scalars as input.
+    const E_ZERO_SCALARS: u64 = 3;
+
+    //
+    // Scalar and point structs
+    //
+
+    /// This struct represents a scalar as a little-endian byte encoding of an integer in $\mathbb{Z}_\ell$, which is
+    /// stored in `data`. Here, \ell denotes the order of the scalar field (and the underlying elliptic curve group).
+    struct Scalar has key, copy, store, drop {
+        data: vector<u8>
+    }
+
+    /// This struct represents a serialized point on the Ristretto255 curve, in 32 bytes.
+    /// This struct can be decompressed from storage into an in-memory RistrettoPoint, on which fast curve arithmetic
+    /// can be performed.
+    struct CompressedRistretto has copy, store, drop {
+        data: vector<u8>
+    }
+
+    /// This struct represents an in-memory Ristretto255 point and supports fast curve arithmetic.
+    ///
+    /// An important invariant: There will never be two RistrettoPoint's constructed with the same handle. One can have
+    /// immutable references to the same RistrettoPoint, of course.
+    struct RistrettoPoint has drop {
+        handle: u64
+    }
+
+    //
+    // Functions for arithmetic on points
+    //
+
+    /// Returns the identity point as a CompressedRistretto.
+    public fun point_identity_compressed(): CompressedRistretto {
+        CompressedRistretto {
+            data: x"0000000000000000000000000000000000000000000000000000000000000000"
```

Comment:
oh , you do know about that....

Reply:
Aren't humans so weird, inconsistent and unpredictable? But lovable nonetheless... 🤗